core_backend/tests/api/test_manage_tags.py

```diff
@@ -0,0 +1,149 @@
+        fullaccess_token: str,
+    ) -> None:
+        response = client.post(
+            "/tag",
+            headers={"Authorization": f"Bearer {fullaccess_token}"},
+            json={
+                "tag_name": tag_name,
+            },
+        )
+        assert response.status_code == 200
+        json_response = response.json()
+        assert "tag_id" in json_response
+
+        response = client.delete(
+            f"/tag/{json_response['tag_id']}",
+            headers={"Authorization": f"Bearer {fullaccess_token}"},
+        )
+
+        assert response.status_code == 200
+
+    @pytest.mark.parametrize(
+        "tag_name",
+        [
+            ("tag_first"),
+            ("tag_second"),
+        ],
+    )
+    async def test_edit_and_retrieve_tag(
+        self,
+        client: TestClient,
+        existing_tag_id: int,
+        tag_name: str,
+        fullaccess_token: str,
+    ) -> None:
+        response = client.put(
+            f"/tag/{existing_tag_id}",
+            headers={"Authorization": f"Bearer {fullaccess_token}"},
+            json={
+                "tag_name": tag_name,
+            },
+        )
+
+        assert response.status_code == 200
+
+        response = client.get(
+            f"/tag/{existing_tag_id}",
+            headers={"Authorization": f"Bearer {fullaccess_token}"},
+        )
+        assert response.status_code == 200
+        assert response.json()["tag_name"] == tag_name
+
+    async def test_edit_tag_not_found(
+        self, client: TestClient, fullaccess_token: str
+    ) -> None:
+        response = client.put(
+            "/tag/12345",
+            headers={"Authorization": f"Bearer {fullaccess_token}"},
+            json={
+                "tag_name": "tag",
+            },
+        )
+
+        assert response.status_code == 404
+
+    async def test_list_tag(
+        self,
+        client: TestClient,
+        existing_tag_id: int,
+        fullaccess_token: str,
+    ) -> None:
+        response = client.get(
+            "/tag", headers={"Authorization": f"Bearer {fullaccess_token}"}
+        )
+        assert response.status_code == 200
+        assert len(response.json()) > 0
+
+    async def test_delete_tag(
+        self, client: TestClient, existing_tag_id: int, fullaccess_token: str
+    ) -> None:
+        response = client.delete(
+            f"/tag/{existing_tag_id}",
+            headers={"Authorization": f"Bearer {fullaccess_token}"},
+        )
+        assert response.status_code == 200
+
+
+async def test_convert_record_to_schema() -> None:
+    tag_id = 1
+    user_id = 123
+    record = TagDB(
+        tag_id=tag_id,
+        user_id=user_id,
+        tag_name="tag",
+        created_datetime_utc=datetime.datetime.utcnow(),
+        updated_datetime_utc=datetime.datetime.utcnow(),
+    )
+    result = _convert_record_to_schema(record)
+    assert result.tag_id == tag_id
+    assert result.user_id == user_id
+    assert result.tag_name == "tag"
```

Comment:
Can you also add a set of tests like `TestMultUserManageContent` to make sure users don't have access to each others' tag tables